geonode/resource/manager.py

```diff
@@ -447,23 +447,21 @@ def copy(self, instance: ResourceBase, /, uuid: str = None, owner: settings.AUTH
                     _resource = copy.copy(instance.get_real_instance())
                     _resource.pk = _resource.id = None
                     _resource.uuid = uuid or str(uuid4())
+                    _resource.save()
                     if isinstance(instance.get_real_instance(), Document):
-                        _resource.save()
                         for resource_link in DocumentResourceLink.objects.filter(document=instance.get_real_instance()):
                             _resource_link = copy.copy(resource_link)
                             _resource_link.pk = _resource_link.id = None
                             _resource_link.document = _resource.get_real_instance()
                             _resource_link.save()
                     if isinstance(instance.get_real_instance(), Dataset):
                         _resource.name = f'{_resource.name}_{uuid1().hex[:8]}'
```

Comment:
@afabiani the  name updated here is not going to be saved

Reply:
Thanks @marthamareal pushed a fix and test case for this one